src/policy_evaluator/policy_evaluator_builder.rs

```diff
@@ -203,12 +176,88 @@ impl PolicyEvaluatorBuilder {
         Ok(())
     }
 
+    fn validate_user_input_build_evaluator(&self) -> Result<()> {
+        self.validate_user_input_build_evaluator_pre()?;
+
+        if self.eval_ctx.is_none() {
+            return Err(anyhow!("Must provide an evaluation context"));
+        }
+
+        Ok(())
+    }
+
+    /// Create the instance of `PolicyEvaluatorPre` to be used
+    pub fn build_pre(&self) -> Result<PolicyEvaluatorPre> {
+        self.validate_user_input_build_evaluator_pre()?;
+
+        let engine = self.build_engine()?;
+        let module = self.build_module(&engine)?;
+
+        let execution_mode = self.execution_mode.unwrap();
+
+        let stack_pre = match execution_mode {
+            PolicyExecutionMode::KubewardenWapc => {
+                let wapc_stack_pre = wapc::StackPre::new(engine, module, self.epoch_deadlines)?;
+                StackPre::from(wapc_stack_pre)
+            }
+            PolicyExecutionMode::Wasi => {
+                let wasi_stack_pre = wasi_cli::StackPre::new(engine, module, self.epoch_deadlines)?;
+                StackPre::from(wasi_stack_pre)
+            }
+            PolicyExecutionMode::Opa | PolicyExecutionMode::OpaGatekeeper => {
+                let rego_stack_pre = rego::StackPre::new(
+                    engine,
+                    module,
+                    self.epoch_deadlines,
+                    0, // currently the entrypoint is hard coded to this value
+                    execution_mode.try_into()?,
+                );
+                StackPre::from(rego_stack_pre)
+            }
+        };
+
+        Ok(PolicyEvaluatorPre { stack_pre })
+    }
+
     /// Create the instance of `PolicyEvaluator` to be used
+    // TODO: later on we can drop this method and force also kwctl to use `build_pre`
```

Comment:
shouldn't we drop this in this PR?

Reply:
Done via 8752bbda6a1599a7ed877735bed64fb50ff0c643